File: Arrays/Intro/EvenNumberDigits.js
// https://leetcode.com/explore/learn/card/fun-with-arrays/521/introduction/3237/

/**
 * @param {number[]} nums
 * @return {number}
 */
var findNumbers = function (nums) {
  return nums
    .reduce((cnt, num) => {
      return `${num}`.length % 2 == 0 ? ++cnt : cnt;
    }, 0)
};

File: Arrays/Intro/EvenNumberDigits.cpp
class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int evens = 0, even = 1;
        for(auto el: nums){
            even = 1;
            while(el){
                el /= 10;
                even = !even;
            }
            if(even) evens++;
        }
        return evens;
    }
};